addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/handler/BridgeHandler.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.icloud.handler;
+
+import static org.openhab.binding.icloud.BindingConstants.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.icloud.discovery.DeviceDiscovery;
+import org.openhab.binding.icloud.internal.Connection;
+import org.openhab.binding.icloud.internal.DeviceInformationParser;
+import org.openhab.binding.icloud.internal.configuration.AccountThingConfiguration;
+import org.openhab.binding.icloud.internal.json.icloud.DeviceInformation;
+import org.openhab.binding.icloud.internal.json.icloud.JSONRootObject;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Retrieves the data for a given account from iCloud and passes the
+ * information to {@link DeviceDiscover} and to the {@link DeviceHandler}s.
+ *
+ * @author Patrik Gfeller - Initial Contribution
+ * @author Hans-Jörg Merk
+ */
+public class BridgeHandler extends BaseBridgeHandler {

```

Comment:
Note that `BridgeHandler` is already an interface in ESH that BaseBridgeHandler implements - so the name of this class can easily cause confusion. Maybe better refactor it to be `ICloudHandler`.

Reply:
Renamed the handler classes to prevent confusion with ESH classes; done.